app/helpers/automated_tests_helper.rb

```diff
@@ -480,6 +486,435 @@ def parse_test_output(repo_assignment_dir, assignment, logfile, data)
+    test_dir = File.join(MarkusConfigurator.markus_config_automated_tests_repository, @assignment.short_identifier)
+    src_dir = @repo_dir
+    assign_dir = @repo_dir + "/" + @assignment.repository_folder
+    if !(File.exists?(test_dir))
+      # TODO: show the error to user instead of raising a runtime error
+      raise I18n.t("automated_tests.test_files_unavailable")
+    elsif !(File.exists?(src_dir))
+      # TODO: show the error to user instead of raising a runtime error
+      raise I18n.t("automated_tests.source_files_unavailable")
+    end
+
+    if !(File.exists?(assign_dir))
+      # TODO: show the error to user instead of raising a runtime error
+      raise I18n.t("automated_tests.source_files_unavailable")
+    end
+
+    dir_contents = Dir.entries(assign_dir)
+
+    #if there are no files in repo (ie only the current and parent directory pointers)
+    if (dir_contents.length <= 2)
+      raise I18n.t("automated_tests.source_files_unavailable")
+    end
+
+    scripts = TestScript.where(assignment_id: @assignment.id)
+    if scripts.empty?
+      # TODO: show the error to user instead of raising a runtime error
+      raise I18n.t("automated_tests.test_files_unavailable")
+    end
+
+    return true
+  end
+
+  # Verify the user has the permission to run the tests - admin
+  # and graders always have the permission, while student has to
+  # belong to the group, and have at least one token.
+  def self.has_permission?()
+    if @current_user.admin?
+      true
+    elsif @current_user.ta?
+      true
+    elsif @current_user.student?
+      # Make sure student belongs to this group
+      if not @current_user.accepted_groupings.include?(@grouping)
+        # TODO: show the error to user instead of raising a runtime error
+        raise I18n.t("automated_tests.not_belong_to_group")
+      end
+      #can skip checking tokens if we have unlimited
+      if @grouping.assignment.unlimited_tokens
+        return true
+      end
+      t = @grouping.token
+      if t == nil
+        raise I18n.t('automated_tests.missing_tokens')
+      end
+      if t.tokens > 0
+        t.decrease_tokens
+        true
+      else
+        # TODO: show the error to user instead of raising a runtime error
+        raise I18n.t("automated_tests.missing_tokens")
+      end
+    end
+  end
+
+
+  # Perform a job for automated testing. This code is run by
+  # the Resque workers - it should not be called from other functions.
+  def self.perform(grouping_id, call_on)
+    # Pick a server, launch the Test Runner and wait for the result
+    # Then store the result into the database
+
+    #@submission = Submission.find(submission_id)
+    @grouping = Grouping.find(grouping_id)
+    @assignment = @grouping.assignment
+    @group = @grouping.group
+    @repo_dir = File.join(MarkusConfigurator.markus_config_automated_tests_repository, @group.repo_name)
+
+    @list_of_servers = MarkusConfigurator.markus_ate_test_server_hosts.split(' ')
```

Comment:
Yeah, I don't want to see any mention of external servers at this point. What I really want is to see the tests run successfully on the server.